### PR TITLE
TEST/UCP: Allow overriding endpoint and worker parameters at runtime.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4499/ for details (Mellanox internal link).

**Comment by MattBBaker** [comment]
bot:ornl:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4502/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4510/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4514/ for details (Mellanox internal link).

File: test/gtest/ucp/test_ucp_wakeup.cc [hunk 1]
```diff
@@ -346,22 +344,10 @@ test_ucp_wakeup_events::do_tx_rx_events_test(const std::vector<std::string>& tra
...
-
 UCS_TEST_P(test_ucp_wakeup_events, events)
 {
     UCS_TEST_SKIP_R("Functionality is not implemented yet");
-
-    EXPECT_TRUE(GetParam().worker_params.field_mask &
-                UCP_WORKER_PARAM_FIELD_EVENTS);
-    do_tx_rx_events_test(GetParam().transports,
-                         GetParam().worker_params.events);
+    // do_tx_rx_events_test(GetParam().transports, GetParam().variant);
```
Comment (evgeny-leksikov) [inline test/gtest/ucp/test_ucp_wakeup.cc]: Commented code easy to miss during refactoring and hardly restore later. Pls uncomment this or remove the test case if not needed anymore.

File: test/gtest/ucp/test_ucp_peer_failure.cc [hunk 2]
```diff
@@ -98,7 +99,7 @@ void test_ucp_peer_failure::init() {
     /* Make second pair */
     create_entity(true);
     create_entity(false);
-    sender().connect(&receiver(), &GetParam().ep_params_cmn);
+    sender().connect(&receiver(), get_ep_params());
```
Comment (evgeny-leksikov) [inline test/gtest/ucp/test_ucp_peer_failure.cc]: which `get_ep_params` implementation will be called here? I guess some one from `test_ucp_tag` parent branch. If so, error handling will be disabled. It looks like need to add `test_ucp_peer_failure::get_ep_params`

File: test/gtest/ucp/test_ucp_wakeup.cc (hunk 1 above)
Comment (yosefe) [reply test/gtest/ucp/test_ucp_wakeup.cc]: actually, you added this test which is always skipped, and coverity started complaining about this.
can we remove the whole test or need to un-skip it?

File: test/gtest/ucp/test_ucp_peer_failure.cc (hunk 2 above)
Comment (yosefe) [reply test/gtest/ucp/test_ucp_peer_failure.cc]: 10x, will fix

File: test/gtest/ucp/test_ucp_wakeup.cc (hunk 1 above)
Comment (evgeny-leksikov) [reply test/gtest/ucp/test_ucp_wakeup.cc]: we don't have implementation to un-skip, lets remove then add new one when feature is implemented.

Comment (yosefe) [reply test/gtest/ucp/test_ucp_wakeup.cc]: ok

File: test/gtest/ucp/test_ucp_peer_failure.cc (hunk 2 above)
Comment (yosefe) [reply test/gtest/ucp/test_ucp_peer_failure.cc]: @evgeny-leksikov there already is `test_ucp_peer_failure::get_ep_params` - in line 19

Comment (evgeny-leksikov) [reply test/gtest/ucp/test_ucp_peer_failure.cc]: This is `test_ucp_peer_failure_base::get_ep_params`, another branch of multi-inheritance.

Comment (yosefe) [reply test/gtest/ucp/test_ucp_peer_failure.cc]: that is why we have `using test_ucp_peer_failure_base::get_ep_params;` in line 58. 
without this, even the compiler complains about ambiguity

Comment (evgeny-leksikov) [reply test/gtest/ucp/test_ucp_peer_failure.cc]: ahh.. right, sorry missed it.

**Comment by yosefe** [comment]
bot:mlnx:retest

**Pull request merged**